cloudify_awssdk/s3/resources/bucket_object.py

```diff
@@ -0,0 +1,164 @@
+from botocore.exceptions import ClientError
+from cloudify.exceptions import NonRecoverableError
+
+# Local Imports
+from cloudify_awssdk.common import decorators, utils
+from cloudify_awssdk.s3 import S3Base
+from cloudify_awssdk.common.constants import EXTERNAL_RESOURCE_ID
+
+RESOURCE_TYPE = 'S3 Bucket Object'
+BUCKET = 'Bucket'
+OBJECT_KEY = 'Key'
+BUCKET_TYPE = 'cloudify.nodes.aws.s3.Bucket'
+
+
+class S3BucketObject(S3Base):
+    """
+        AWS S3 Bucket Object interface
+    """
+    def __init__(self, ctx_node, resource_id=None, client=None, logger=None):
+        S3Base.__init__(self, ctx_node, resource_id, client, logger)
+        self.type_name = RESOURCE_TYPE
+        self.resource_config = self.get_resource_config(ctx_node)
+        self._bucket_name = self.resource_config[BUCKET] if \
+            self.resource_config.get(BUCKET) else None
+
+    def get_resource_config(self, node):
+        return node.properties['resource_config']['kwargs'] if \
+            node.properties.get('resource_config') \
+            and node.properties['resource_config'].get('kwargs') else None
+
+    @property
+    def bucket_name(self):
+        return self._bucket_name
+
+    @bucket_name.setter
+    def bucket_name(self, value):
+        self._bucket_name = value
+
+    @property
+    def properties(self):
+        """Gets the properties of an external resource"""
+        resource = None
+        try:
+            resource = \
+                self.client.head_object(
+                    **{OBJECT_KEY: self.resource_id,
+                       BUCKET: self.bucket_name})
+        except ClientError:
+            pass
+
+        return resource
+
+    @property
+    def status(self):
+        """Gets the status of an external resource"""
+        if self.properties:
+            return 'available'
+        return None
+
+    def create(self, params):
+        """
+            Create a new AWS S3 Bucket Object.
+        """
+        self.logger.debug('Creating {0} with parameters: {1}'
+                          .format(self.type_name, params))
+        res = self.client.upload_file(**params)
+        return res
+
+    def delete(self, params=None):
+        """
+            Deletes an existing AWS S3 Bucket Object.
+        """
+        self.logger.debug('Deleting {0} with parameters: {1}'
+                          .format(self.type_name, params))
+        self.client.delete_object(**params)
+
+
+@decorators.aws_resource(resource_type=RESOURCE_TYPE)
+def prepare(ctx, resource_config, **_):
+    """Prepares an AWS S3 Bucket Object"""
+    # Save the parameters
+    ctx.instance.runtime_properties['resource_config'] = resource_config
+
+
+@decorators.aws_resource(S3BucketObject, RESOURCE_TYPE)
+def create(ctx, iface, resource_config, **_):
+    """Creates an AWS S3 Bucket Object"""
+
+    # Create a copy of the resource config for clean manipulation.
+    params = \
+        dict() if not resource_config else resource_config.copy()
+
```

Comment:
I believe that we will need to have some special handling using `download_resource` when using a manager since the execution directory on a manager is not the blueprint root.

Reply:
@EarthmanT I did update the code to add support to download file using dowload_resource